app/src/org/commcare/android/database/app/DatabaseAppOpenHelper.java

```diff
@@ -67,19 +73,24 @@ public void onCreate(SQLiteDatabase database) {
             builder = new TableBuilder(UserKeyRecord.class);
             database.execSQL(builder.getTableCreateString());
 
-            database.execSQL("CREATE INDEX global_index_id ON GLOBAL_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
-            database.execSQL("CREATE INDEX upgrade_index_id ON UPGRADE_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
-            database.execSQL("CREATE INDEX recovery_index_id ON RECOVERY_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
+            database.execSQL(tableIndexQuery("GLOBAL_RESOURCE_TABLE", "global_index_id"));
+            database.execSQL(tableIndexQuery("UPGRADE_RESOURCE_TABLE", "upgrade_index_id"));
+            database.execSQL(tableIndexQuery("RECOVERY_RESOURCE_TABLE", "recovery_index_id"));
+            database.execSQL(tableIndexQuery(AndroidResourceManager.TEMP_UPGRADE_TABLE_KEY, "temp_upgrade_index_id"));
 
             DbUtil.createNumbersTable(database);
 
             database.setTransactionSuccessful();
-
         } finally {
             database.endTransaction();
         }
     }
 
+    public static String tableIndexQuery(String tableName, String indexName) {
+        return "CREATE INDEX " + indexName + " ON " +
+                tableName + "( " + Resource.META_INDEX_PARENT_GUID + " )";
```

Comment:
Sounds good: I renamed `tableIndexQuery` to `indexOnTableWithPGUIDCommand` and also created the helper function `indexOnTableCommand` to take in column args.